octopusdeploy/resource_tenant.go

```diff
@@ -24,6 +24,9 @@ func resourceTenant() *schema.Resource {
 }
 
 func resourceTenantCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	mutex.Lock()
```

Comment:
I hadn't seen this pattern of locking the resource interaction behind a mutex. I see this error a lot:

```
Error: Octopus API error: There was a problem with your request. [Changes to these variables could not be saved, because another user has made changes to the variables between when you started editing and when you saved your changes. Please reload or open a new tab to make your changes.]
```

I'm thinking variables might have to use a mutex as well.